data/docs/gcp-monitoring/gke-tracing.mdx

```diff
@@ -0,0 +1,1190 @@
+<figcaption>
+<i>
+Pixie Configuration
+</i>
+</figcaption>
+</figure>
+
+
+**Step 3:** To test the configuration, deploy a sample application in the cluster to view its traces in the SigNoz.
+
+```sh
+kubectl run -n signoz my-hello –image=nginx –port=80
+```
+
+### Visualize the Traces in SigNoz self-hosted
+
+To visualize the traces, log into the SigNoz account and navigate to the traces section.
+
+<Tabs>
+<TabItem value="self-hosted-on-gke-cluster" label="For self-hosted SigNoz in GKE" default>
+
+<figure data-zoomable align="left">
+    <img
+        src="/img/docs/gcp-monitoring/gke-tracing/signoz-self-hosted-gke-ebpf.webp"
+        alt="Traces in SigNoz Dashboard"
+    />
+<figcaption>
+<i>
+Traces in SigNoz Dashboard (self-hosted on GKE Cluster)
+</i>
+</figcaption>
+</figure>
+
+</TabItem>
+
+<TabItem value="self-host-signoz-in-vm" label="For self-hosted SigNoz in VM">
+
+<figure data-zoomable align="left">
+    <img
+        src="/img/docs/gcp-monitoring/gke-tracing/signoz-sel-hosted-ebpf.webp"
+        alt="Traces in SigNoz Dashboard"
+    />
+<figcaption>
+<i>
+Traces in SigNoz Dashboard (self-hosted on a VM)
+</i>
+</figcaption>
+</figure>
+
+</TabItem>
+</Tabs>
+
+## APM and Distributed Tracing
+
+For application-level tracing, you can use the OpenTelemetry SDKs integrated with your application. These SDKs automatically collect and forward traces to the central collector.
+
+Please refer to our [SigNoz Tutorials](https://signoz.io/docs/instrumentation/) or [Blog](https://signoz.io/blog/) to find information on how to instrument your application like Spring, FastAPI, NextJS, Langchain, Node.js, Flask, Django, etc.
+
+### Sample Python Application 
+
+We will use a [sample flask app](https://github.com/SigNoz/sample-flask-app).
+One may need to add packages (in `requirements.txt`) for enabling instrumentation.
+
+```txt
+Flask==3.0.0 
+pymongo==3.12.1 
+requests==2.26.0 
+opentelemetry-api==1.22.0 
+opentelemetry-distro==0.43b0 
+opentelemetry-instrumentation==0.43b0 
+opentelemetry-exporter-otlp==1.22.0 
+``` 
+
+Install those packages in dockerfile configuration.
+
+```Dockerfile 
+… 
+# install dependencies
+RUN pip install -r requirements.txt
+RUN opentelemetry-bootstrap --action=install…  
+``` 
+ 
+### Example YAML 
+For running this sample-flask-app as pod,use the following yaml file.  
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  creationTimestamp: null
+  labels:
+    app: sample-flask-app
+  name: sample-flask-app
+spec:
+  containers:
+  - image: mongo:latest
+    name: mongo
+    ports:
+    - name: mongo
+      containerPort: 27017
+  - image: signoz/sample-flask-app:latest
+    name: sample-app
+    ports:
+    - name: flask
+      containerPort: 5002
+    env:
+    - name: MONGO_HOST
+      value: localhost
+    - name: OTEL_RESOURCE_ATTRIBUTES
+      value: service.name=sample-app
+    - name: OTEL_EXPORTER_OTLP_ENDPOINT
+      value: http://kubelet-otel-k8s-infra-otel-agent.signoz.svc.cluster.local:4317
```

Comment:
Need to change this to  <IP Address of the VM>:4317

Reply:
I think we didn't mention it right on the doc; I converted from that, so it better correct it there as well.

Reply:
fixed